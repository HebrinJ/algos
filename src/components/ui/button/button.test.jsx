import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, fireEvent, screen } from '@testing-library/react';

it('Should render default button with text', () => {
    const tree = renderer
    .create(<Button text='Нажми меня' isLoader={false} disabled={false} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

it('Should renderer default button without text', () => {
    const tree = renderer
    .create(<Button text='' isLoader={false} disabled={false} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

it('Should renderer button with loader', () => {
    const tree = renderer
    .create(<Button text='Лоадер' isLoader={true} disabled={false} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

it('Should renderer disabled button', () => {
    const tree = renderer
    .create(<Button text='Отключенная кнопка' isLoader={false} disabled={true} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

it('Should call onClick handler', () => {
    const mockFn = jest.fn();
    
    render(<Button text='Нажми меня' isLoader={false} disabled={false} onClick={mockFn}/>);
    const button = screen.queryByText('Нажми меня')
    fireEvent.click(button);

    expect(mockFn).toHaveBeenCalledTimes(1);
}); 