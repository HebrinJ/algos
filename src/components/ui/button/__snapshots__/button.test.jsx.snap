// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render default button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={false}
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня
  </p>
</button>
`;

exports[`Should renderer button with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Should renderer default button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={false}
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Should renderer disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Отключенная кнопка
  </p>
</button>
`;
